pipeline/transformer/transformer_miro/src/main/scala/uk/ac/wellcome/platform/transformer/miro/services/MiroVHSRecordReceiver.scala

```diff
@@ -27,19 +41,19 @@ class MiroVHSRecordReceiver(objectStore: ObjectStore[MiroRecord],
                        Int) => Try[TransformedBaseWork]): Future[Unit] = {
     debug(s"Starting to process message $message")
 
-    val futurePublishAttempt = for {
-      hybridRecord <- Future.fromTry(fromJson[HybridRecord](message.body))
-      miroMetadata <- Future.fromTry(fromJson[MiroMetadata](message.body))
-      transformableRecord <- getTransformable(hybridRecord)
-      work <- Future.fromTry(
-        transformToWork(transformableRecord, miroMetadata, hybridRecord.version)
-      )
-      publishResult <- publishMessage(work)
-      _ = debug(
-        s"Published work: ${work.sourceIdentifier} with message $publishResult")
-    } yield publishResult
-
-    futurePublishAttempt
+    val msgNotification = Future.fromTry {
+      for {
+        record <- fromJson[HybridRecord](message.body)
+        (miroRecord, miroMetadata) <- getRecordAndMetadata(
+          Version(record.id, record.version))
```

Comment:
Rather than exposing the `Version` interface up here (which is really a detail of Store), perhaps move it inside `getRecordAndMetadata`?)